### ISSUE TITLE
New release, 2.0.1, 2.1.0?

### ISSUE THREAD

**Issue by bungle** [lunarmodules/busted#668]
Several things have been added to master. Could we get a new release of busted?

**Comment by Frenzie** [comment]
A new release with this line changed to `https://` would be most welcome.
https://github.com/Olivine-Labs/busted/blob/75b2c8930ed22d99dc5f6e41547ec333feee6197/busted-scm-2.rockspec#L4

```
busted scm-2 depends on mediator_lua >= 1.1.1 (not installed)
Installing https://luarocks.org/mediator_lua-1.1.2-0.rockspec
mediator_lua 1.1.2-0 is now installed in /usr/local (license: MIT <http://opensource.org/licenses/MIT>)

Cloning into 'busted'...
fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.

Error: Failed cloning git repository.
Error: Process completed with exit code 1.
```

**Comment by Tieske** [comment]
@DorianGray @ajacksified The last release was almost 4 years ago. Seems like you have moved on and/or changed priorities.

Would it be possible to transfer the repositories for [busted](https://github.com/Olivine-Labs/busted), [luassert](https://github.com/Olivine-Labs/luassert), and [say](https://github.com/Olivine-Labs/say), to the [lunarmodules](https://github.com/lunarmodules) organization? This would allow us to set up a new team and keep the project going forward.

**Comment by alerque** [comment]
I second that suggestion! Busted and friends are *staples* in the Lua ecosystem and deserve good active maintenance. I'm willing to pitch in some here and there as I have for other projects hosted there (Luacheck, LDoc, Penlight, etc.).

**Comment by ajacksified** [comment]
@Tieske I've checked in with @DorianGray and we're happy to transfer the repositories.

**Comment by ajacksified** [comment]
Done - I'll just need to set up a redirect for the docs.

**Comment by ajacksified** [comment]
Also, let me know if there's anything I need to do to get you access for luarocks or anything else.

**Comment by alerque** [comment]
@ajacksified The publishing to LuaRocks could me handled either of two ways:

1. You or whoever has access to the current [olivine-labs](https://luarocks.org/modules/olivine-labs) user could add an API key to that account and add it as a secret to these 3 repos.

1. We can ask Hishamh to copy the module into the [lunarmodules](https://luarocks.org/modules/lunarmodules) user which has an API secret this whole org can use to publish already.

I don't feel too strongly either way, perhaps @Tieske or others do. Both ways cause some end user confusion about who is who and what is what but both work in the end. A little bit of history gets lost when copying to a different user on LuaRocks, for example the release dates of old versions. Also the old namespace is still there so people following direct links can end up in the wrong place even if the package they get from `luarocks install ...` is the new package.

**Comment by alerque** [comment]
Note we already have some projects that have kept their previous publishers (e.g. Penlight) and some that have switched to Lunarmodules (e.g. Luacheck) so there really isn't a precedent. Whatever you guys think is best for the project...

**Comment by Tieske** [comment]
If we don’t want to bother @ajacksified or @DorianGray  again, then I think transferring the luarocks user  to Lunarmodules makes the most sense to me. The lost history is a temporary thing anyway… my 2cts

**Comment by ajacksified** [comment]
Cool. What's the best way to go about that - should I ping @hishamhm?

**Comment by hishamhm** [comment]
Done, [busted in luarocks.org is owned by the lunarmodules account now](https://luarocks.org/modules/lunarmodules/busted)

**Comment by ajacksified** [comment]
Thanks!

**Comment by Tieske** [comment]
Just noticed @alerque tagged it, so version `2.0` is up!

**Comment by alerque** [comment]
Actually I didn't tag it. The v2.O.0 tag already existed from 2019. The only thing I did was mark the tag as being a GH Release. The issue I saw was all the other tags, notably over a dozen RC tags, were all marked as being GH releases but the final 2.0.0 was not. That was causing the GH repo to show RC13 or something like that as the "latest" release.

Unfortunately the GH interface shows the release date as when I marked it not the date of the tag that was marked, but if you click through to the release you'll see which commit is tagged and that the tag is from June 28, 2019.

I would suggest we cleanup a few things in the repo like links and such, get the GH Action configuration working (see #689), get the project copied to the lunarmodules namespace on luarocks.org, then cut a 2.1 release with the 50ish commits since the last tagging.

**Comment by hishamhm** [comment]
I got notifications that `luassert` and `say` have been moved to lunarmodules as well on Github. I assume I should move those too on luarocks.org?

**Comment by ajacksified** [comment]
Yes please!

**Comment by hishamhm** [comment]
done!

**Comment by alerque** [comment]
Thanks @hishamhm.

BTW I nixed getting this repo to work as a GH action from my list at least for this cycle. It's quite possible to do but it will take a lot more fiddling to setup in a way that is useful to many project than the easy way I used for Luacheck, so I don't think that should hold up a release.

I think #689 should cover most of point 1. Otherwise we have some fiddly bits like making sure the changelog is sane and reflects the 3 years of commits.

**Comment by alerque** [comment]
I've posted PRs updating links and CI release related stuff for [say](https://github.com/lunarmodules/say/pull/16), [luassert](https://github.com/lunarmodules/luassert/pull/189), and [busted](https://github.com/lunarmodules/busted/pull/689). All of these three projects have minor improvements in commits since the last release. My personal idea at this point would be to merge all of those, then work through them *in that order* doing minor-version release bumps for each and making sure the automatic publishing stuff works.

I have tagged @Tieske as a potential reviewer on all three, but honestly I'd move forward with that plan if *anybody* in @lunarmodules gave it a skim and review approval for each.

These would basically be safe-harbor release representing the current Git HEAD for each project prior to migration to this org plus our release tooling buts. We could then look into further triaging open issues & PRs.
